digsum.rb

```diff
@@ -0,0 +1,16 @@
+def digsum_r(n)
+  if n < 10
+    n
+  else
+    n % 10 + digsum_r(n/10)
+  end
+end
+
+def digsum_i(n)
+  sum = 0
+  while n > 0
+    sum += n % 10
+    n /= 10
```

Comment:
Thank you for leading me to the light on the digsum recursion. It's like working out a muscle group of the mind I'm not used to working out.